xbmc/games/dialogs/osd/DialogGameSaves.h

```diff
@@ -36,7 +36,7 @@ class CDialogGameSaves : public CGUIDialogSelect
   /*!
    * \brief Called every frame with the item being focused
    */
-  void OnFocus(CFileItemPtr item);
+  void OnFocus(const CFileItemPtr& item);
```

Comment:
Better would be `void OnFocus(const CFileItem& item);`. Change the signature to dereference the pointer to remove the possibility of a null pointer dereference in the inner function. You check `if (item)` outside the function so that inside the function you can always guarantee a valid object and safely remove all inner pointer dereferences.